docs/how-to/react-server-components.md

```diff
@@ -1,50 +1,738 @@
+  })
+);
+// Hook up our application.
+app.use(
+  createRequestListener((request) =>
+    generateHTML(
+      request,
+      fetchServer,
+      (routes as unknown as { bootstrapScript?: string }).bootstrapScript
+    )
+  )
+);
+
+app.listen(3000, () => {
+  console.log("Server listening on port 3000");
+});
+```
+
+#### `entry.client.tsx`
+
+```tsx filename=src/entry.client.tsx
+"use client-entry";
+
+import { startTransition, StrictMode } from "react";
+import { hydrateRoot } from "react-dom/client";
+import {
+  unstable_createCallServer as createCallServer,
+  unstable_getRSCStream as getRSCStream,
+  unstable_RSCHydratedRouter as RSCHydratedRouter,
+  type unstable_RSCPayload as RSCServerPayload,
+} from "react-router";
+import {
+  createFromReadableStream,
+  createTemporaryReferenceSet,
+  encodeReply,
+  setServerCallback,
+} from "react-server-dom-parcel/client";
+
+// Create and set the callServer function to support post-hydration server actions.
+setServerCallback(
+  createCallServer({
+    createFromReadableStream,
+    createTemporaryReferenceSet,
+    encodeReply,
+  })
+);
+
+// Get and decode the initial server payload.
+createFromReadableStream(getRSCStream()).then(
+  (payload: RSCServerPayload) => {
+    startTransition(async () => {
+      const formState =
+        payload.type === "render"
+          ? await payload.formState
+          : undefined;
+
+      hydrateRoot(
+        document,
+        <StrictMode>
+          <RSCHydratedRouter
+            createFromReadableStream={
+              createFromReadableStream
+            }
+            payload={payload}
+          />
+        </StrictMode>,
+        {
+          formState,
+        }
+      );
+    });
+  }
+);
+```
+
+### Vite
+
+See the [Vite RSC docs][vite-rsc-doc] for more information.
+
+In addition to `react`, `react-dom`, and `react-router`, you'll need the following dependencies:
+
+```shellscript
+npm i -D vite @vitejs/plugin-react @vitejs/plugin-rsc
+```
+
+#### `vite.config.ts`
+
+To configure Vite, add the following to your `vite.config.ts`:
+
+```ts filename=vite.config.ts
+import rsc from "@vitejs/plugin-rsc/plugin";
+import react from "@vitejs/plugin-react";
+import { defineConfig } from "vite";
+
+export default defineConfig({
+  plugins: [
+    react(),
+    rsc({
+      entries: {
+        client: "src/entry.client.tsx",
+        rsc: "src/entry.rsc.tsx",
+        ssr: "src/entry.ssr.tsx",
+      },
+    }),
+  ],
+});
+```
+
+#### `entry.ssr.tsx`
+
+The following is a simplified example of a Vite SSR Server.
+
+```tsx filename=src/entry.ssr.tsx
+import { createFromReadableStream } from "@vitejs/plugin-rsc/ssr";
+import { renderToReadableStream as renderHTMLToReadableStream } from "react-dom/server.edge";
+import {
+  unstable_routeRSCServerRequest as routeRSCServerRequest,
+  unstable_RSCStaticRouter as RSCStaticRouter,
+} from "react-router";
+import bootstrapScriptContent from "virtual:vite-rsc/bootstrap-script-content";
```

Comment:
This API has been changed to `const bootstrapScriptContent = await import.meta.viteRsc.loadBootstrapScriptContent("index");`, which should be called in `renderHTML`.

Reply:
Ah great, thanks!